common/layerSharedLogic.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed
+ * under the Apache License Version 2.0.
+ *
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2021 Datadog, Inc.
+ */
+
+import * as crypto from "crypto";
+import log from "loglevel";
+import {
+  runtimeToLayerName,
+  DD_GOV_ACCOUNT_ID,
+  DD_ACCOUNT_ID,
+  LAYER_PREFIX,
+  EXTENSION_LAYER_PREFIX,
+} from "./constants";
+
+export function getLambdaLayerArn(region: string, version: number, runtime: string, isArm: boolean, isNode: boolean) {
+  const baseLayerName = runtimeToLayerName[runtime];
+  const layerName = isArm && !isNode ? `${baseLayerName}-ARM` : baseLayerName;
+  // TODO: edge case where gov cloud is the region, but they are using a token so we can't resolve it.
+  const isGovCloud = region === "us-gov-east-1" || region === "us-gov-west-1";
+
+  // if this is a GovCloud region, use the GovCloud lambda layer
+  if (isGovCloud) {
+    log.debug("GovCloud region detected, using the GovCloud lambda layer");
+    return `arn:aws-us-gov:lambda:${region}:${DD_GOV_ACCOUNT_ID}:layer:${layerName}:${version}`;
+  }
+  return `arn:aws:lambda:${region}:${DD_ACCOUNT_ID}:layer:${layerName}:${version}`;
+}
+
+export function getExtensionLayerArn(region: string, version: number, isArm: boolean) {
+  const baseLayerName = "Datadog-Extension";
+  const layerName = isArm ? `${baseLayerName}-ARM` : baseLayerName;
+  const isGovCloud = region === "us-gov-east-1" || region === "us-gov-west-1";
+  if (isGovCloud) {
+    log.debug("GovCloud region detected, using the GovCloud extension layer");
+    return `arn:aws-us-gov:lambda:${region}:${DD_GOV_ACCOUNT_ID}:layer:${layerName}:${version}`;
+  }
+  return `arn:aws:lambda:${region}:${DD_ACCOUNT_ID}:layer:${layerName}:${version}`;
+}
+
+export function getMissingLayerVersionErrorMsg(functionKey: string, formalRuntime: string, paramRuntime: string) {
+  return (
+    `Resource ${functionKey} has a ${formalRuntime} runtime, but no ${formalRuntime} Lambda Library version was provided. ` +
+    `Please add the '${paramRuntime}LayerVersion' parameter for the Datadog serverless macro.`
+  );
+}
+
+export function generateLambdaLayerId(lambdaFunctionArn: string, runtime: string) {
+  log.debug("Generating construct Id for Datadog Lambda layer");
+  const layerValue: string = crypto.createHash("sha256").update(lambdaFunctionArn).digest("hex");
+  return LAYER_PREFIX + "-" + runtime + "-" + layerValue;
```

Comment:
nit : you can use \`${variable}-text1-${var2}\` for concatenation

Reply:
fixed